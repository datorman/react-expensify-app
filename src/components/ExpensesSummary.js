import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenses from './../selectors/expenses';
import selectExpensesTotal from './../selectors/expenses-total';

export const ExpensesSummary = ({expenseCount,expenseTotal}) => {
    return(
        <div>
            <h3>Viewing {expenseCount} {expenseCount===1 ? 'expense':'expenses'} totaling {numeral(expenseTotal/100).format('$0,0.00')}</h3>
        </div> 
    );   
};


// Not needed but saving for ref
// export const ExpenseSummary = (props) => (
//     <div>
//       {props.expenses.length===0?'': <h3>Viewing {props.expenses.length} expenses totaling {numeral(expensesTotal(props.expenses)/100).format('$0,0.00')}</h3>}
//     </div>   
// );
const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return{
        expenseCount:visibleExpenses.length,
        expenseTotal:selectExpensesTotal(visibleExpenses)
    }
};
export default connect(mapStateToProps)(ExpensesSummary);