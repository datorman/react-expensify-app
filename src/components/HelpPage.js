import React from 'react';
const HelpPage = () => (
    <div>
      test4
    </div>
);
export default HelpPage;