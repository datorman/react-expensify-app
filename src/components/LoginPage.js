import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from './../actions/auth';

export const LoginPage = ({startLogin}) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className = "box-layout__title">Expensify Login</h1>
            <p>Manage Your Expenses</p>
            <button className = "button box-layout__button" onClick = {startLogin}>Login With Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) =>({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);