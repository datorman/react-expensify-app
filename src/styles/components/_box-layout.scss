.box-layout{
    background: url('/images/bg.jpg')no-repeat;
    background-size:cover;
    height:100vh;
    width: 100vw;
    display:flex;
    justify-content: center;
    align-items: center;
}
.box-layout__box{
    background: fade-out(white,.10);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align:center;
    width:25rem;
}
.box-layout__title{
    margin:0 0 $m-size;
    line-height: 1;

}