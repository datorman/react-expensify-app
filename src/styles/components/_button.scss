.button{
    background: $blue;
    color:#fff;
    display:inline-block;
    font-size:$font-size-large;
    padding: $s-size ;
    border:none;
    font-weight:300;
    text-decoration: none;
    line-height:1;
}
.button__link{
    background:none;
}
.button__secondary{
    background: $grey;
}