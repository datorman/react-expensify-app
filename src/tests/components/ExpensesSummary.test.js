import React from 'react';
import {ExpensesSummary} from './../../components/ExpensesSummary';
import {shallow} from 'enzyme';



test('Should render expense total message for 1 item', () =>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={123} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render expense total message for 2 item', () =>{
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expenseTotal={123123123} />);
    expect(wrapper).toMatchSnapshot();
});