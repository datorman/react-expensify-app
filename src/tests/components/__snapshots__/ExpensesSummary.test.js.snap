// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense total message for 1 item 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totaling 
      <span>
        $1.23
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render expense total message for 2 item 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       
      expenses
       totaling 
      <span>
        $1,231,231.23
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
